agents/agent.py

```diff
@@ -399,38 +472,41 @@ def status_to_controller(self, timeout=5, api_key=None):
                 logger=self.logger,
                 fallback_logger=self.fallback_logger,
             )
-
             return
 
-        payload = self.status_to_json(log=False)
+        payload_str = self.status_to_json(log=False)
+        payload = json.loads(payload_str)
 
-        headers = {'Content-Type': 'application/json'}
         if api_key:
-            headers.update({'X-API-Key': api_key})
-
-        request = urllib2.Request(
-            self.controller_url, payload, headers=headers
-        )
-
-        status_code = None
+            payload['api_key'] = api_key
 
         try:
-            response = urllib2.urlopen(request, timeout=timeout)
-            status_code = response.getcode()
-        except (urllib2.URLError, urllib2.HTTPError, socket.timeout) as e:
-            status_code = getattr(e, 'code', None)
-
+            result = self.post_data(self.controller_url, payload)
```

Comment:
Since `status_to_controller` wraps `post_data`, could you add optional arguments `max_retries` and `delay` to `status_to_controller` as well? So that the wrapping function doesn't take away functionality from the wrapped function.